NDTensors/src/tensoralgebra/generic_tensor_operations.jl

```diff
@@ -124,6 +124,10 @@ function _contract!!(
       β,
     )
   else
+    # Special case when ther are unallocated zeros return nothing do no work.
+    if is_unallocated_zeros(tensor1) || is_unallocated_zeros(tensor2)
+      return output_tensor
```

Comment:
One exception I could imagine for this is if the element type of `output_tensor` is `UnspecifiedZero`, while the element types of `tensor1` and/or `tensor2` are specified. Then, I think the element type of `output_tensor` should become specified based on the element types of `tensor1` and `tensor2`. So maybe it would be:
```julia
return set_eltype_if_unspecified(output_tensor, promote_type(eltype(tensor1), eltype(tensor2))
```

I'm not sure if the types of `α` and `β` should play a role, since I think we want that if the element types of `tensor1` and `tensor2` are unspecified, then `output_tensor` would remain unspecified.